Samples/WindowsCopilotRuntime/Contributing.md

```diff
@@ -0,0 +1,92 @@
+  - `cs-winui/ViewModels/ImageDescriptionViewModel.cs`
+  - `cs-winui/ViewModels/mageObjectExtractorViewModel.cs`
+- **Pages**: Contains the XAML pages and their code-behind files. They represent the `Views` part of MVVM.
+  - `cs-winui/Pages/ImageDescriptionPage.xaml`
+  - `cs-winui/Pages/ImageDescriptionPage.xaml.cs`
+  - `cs-winui/Pages/ImageObjectExtractorPage.xaml`
+  - `cs-winui/Pages/ImageObjectExtractorPage.xaml.cs`
+- **Helpers**: Util, Converters and Controls folders contain common shared code used by all the pages
+  - `Controls` folder contains user controls for showing creating Session and example source codes
+  - `CodeBlockControl.xaml` and `CodeBlockControl.xaml.cs` are parts of user control which shows example source code for each api. The example code is fetched from `Examples` folder.
+  - `ModelInitializationControl.xaml` and `ModelInitializationControl.xaml.cs` have user control to create session for every page. Each page sets `CreateModelSessionWithProgress` as `ICommand` to button in this control. 
+  
+## Adding a New Page
+Any of the examples like `ImageScaler` or `ImageDescriptionModel` can be as reference for adding newer pages. `LanguageModel` is slightly different as it has multiple effects in the sample page. 
+
+
+## AsyncCommand and Its Inherited Class
+
+In this project, we use `AsyncCommand` to handle asynchronous operations in the view models. `AsyncCommand` is a custom implementation of `ICommand` that supports asynchronous execution.
+
+### AsyncCommand
+
+Here's an example of how to use `AsyncCommand`:
+
+```csharp
+...
+    public ICommand ScaleCommand => _scaleSoftwareBitmapCommand;
+    private readonly AsyncCommand<(SoftwareBitmap, int, int), SoftwareBitmapSource> _scaleSoftwareBitmapCommand;
+
+    public ImageScalerViewModel(ImageScalerModel imageScalerSession)
+    : base(imageScalerSession)
+    {
+        _scaleSoftwareBitmapCommand = new(
+            async _ =>
+            {
+                var height = (int)(Input!.PixelHeight * Factor);
+                var width = (int)(Input!.PixelWidth * Factor);
+                var outputBitmap = Session.ScaleSoftwareBitmap(Input!, width, height);
+                ...
+            },
+            (_) => IsAvailable && Input is not null);
+    }
+
+...
+
+```
+Here, our code needs to call `ImageScalerModel.ScaleSoftwareBitmap(SoftwareBitmap inputImage, int, int)` from the view model's Scale Image button (refer [`ImageScalerPage.xaml`](./cs-winui/Pages/ImageScalerPage.xaml)). We bind it to `ICommand ScaleCommand` which gets instantiated to `AsyncCommand` (inherited from `ICommand`) and calls `ScaleSoftwareBitmap` like above.
+The `ICommand` is enabled only when Input image is loaded and session is created, denoted by Image and IsAvailable respectively.
+
+### AsyncCommandWithProgress
+
+`AsyncCommandWithProgress` is an inherited class from `AsyncCommand` that supports receiving response in a progressive manner, few tokens at a time.
```

Comment:
```suggestion
`AsyncCommandWithProgress` inherits from `AsyncCommand` and supports receiving response in a progressive manner, a few tokens at a time.
```

Minor: copy edits

Is "token" this right word here? Do all the APIs use the term token for their output?

Reply:
AsyncCommandWithProgress is only used in Language Model, so token is all right.